<?xml version="1.0"?>
<launch>

  <!-- roslaunch arguments -->
<arg name="debug" default="false"/>
<arg name="gui" default="true"/>
<arg name="TAO" default="false"/>
<arg name="robot_ns" default="/"/>
<arg name="robot_name" default="lwr_sim"/>
<!--PTP {a1 0,a2 87,a3 87,a4 0,a5 -55,a6 0,e1 0} -->
<arg name="use_initial_joint_positions" default="false"/>
<arg name="joint_0_init" default="0.0"/>
<arg name="joint_1_init" default="0.0"/>
<arg name="joint_2_init" default="0.0"/>
<arg name="joint_3_init" default="0.0"/>
<arg name="joint_4_init" default="0.0"/>
<arg name="joint_5_init" default="0.0"/>
<arg name="joint_6_init" default="0.0"/>
<arg name="paused" default="false"/>
<arg name="use_sim_time" default="true"/>

<arg name="load_robot_description" default="true"/>
<!-- Tools -->
<arg name="load_base" default="false" />
<arg name="load_ati_sensor" default="true" />
<arg name="load_calib_tool" default="false" />
<arg name="load_handle" default="false" />

<!-- Gazebo has some hardcoded 'waiting for /gazebo... so it should not be in a sub namespace'-->
<!--group ns="$(arg robot_ns)"-->
    <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
    <include file="$(find rtt_lwr_sim)/launch/empty_world.launch">
            <arg name="debug" value="$(arg debug)" />
            <arg name="gui" value="$(arg gui)" />
            <arg name="paused" value="$(arg paused)"/>
            <arg name="use_sim_time" value="$(arg use_sim_time)"/>
            <arg name="headless" value="false"/>
            <arg name="world_name" value="$(find rtt_lwr_sim)/launch/lwr.world"/>
    </include>
    <!--param name="tf_prefix" value="$(arg robot_ns)" /-->
    <!--param name="robot_name" value="$(arg robot_name)"/-->

    <include if="$(arg load_robot_description)" file="$(find lwr_description)/launch/lwr_upload.launch">
        <arg name="load_base" value="$(arg load_base)" />
        <arg name="load_ati_sensor" value="$(arg load_ati_sensor)" />
        <arg name="load_calib_tool" value="$(arg load_calib_tool)" />
        <arg name="load_handle" value="$(arg load_handle)" />
    </include>

    <node ns="$(arg robot_ns)" if="$(arg use_initial_joint_positions)" name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model $(arg robot_name) 
          -J joint_0 $(arg joint_0_init)
          -J joint_1 $(arg joint_1_init)
          -J joint_2 $(arg joint_2_init)
          -J joint_3 $(arg joint_3_init)
          -J joint_4 $(arg joint_4_init)
          -J joint_5 $(arg joint_5_init)
          -J joint_6 $(arg joint_6_init)
          "  respawn="false" output="screen" />
    
    <node ns="$(arg robot_ns)" unless="$(arg use_initial_joint_positions)" name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model $(arg robot_name)"  respawn="false" output="screen" />
    <!-- Launch the CORBA TO naming service -->
    <node if="$(arg TAO)" name="corba_naming_service" pkg="rtt_ros" type="corba-naming-service" args="-m 1"/>

<!--/group-->



</launch>
